client/jetpack-app/plans/main.tsx

```diff
@@ -0,0 +1,113 @@
+import QueryPlans from 'calypso/components/data/query-plans';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { LoadingEllipsis } from 'calypso/components/loading-ellipsis';
+import Main from 'calypso/components/main';
+import { getPlanCartItem } from 'calypso/lib/cart-values/cart-items';
+import PlansFeaturesMain from 'calypso/my-sites/plans-features-main';
+import { getPlanSlug } from 'calypso/state/plans/selectors';
+import { AppState } from 'calypso/types';
+
+import './style.scss';
+
+interface HeaderProps {
+	paidDomainName?: string;
+}
+
+interface JetpackAppPlansProps {
+	paidDomainName?: string;
+	redirectTo?: string;
+}
+
+const Header: React.FC< HeaderProps > = ( { paidDomainName } ) => {
+	const translate = useTranslate();
+
+	return (
+		<div className="plans__header">
+			<FormattedHeader
+				brandFont
+				headerText={ translate( 'Choose the perfect plan' ) }
+				align="center"
+			/>
+			{ paidDomainName ? (
+				<>
+					<p>
+						{ translate(
+							'With your annual plan, you’ll get %(domainName)s {{strong}}free for the first year{{/strong}}.',
+							{
+								args: {
+									domainName: paidDomainName,
+								},
+								components: { strong: <strong /> },
+							}
+						) }
+					</p>
+					<p>
+						{ translate(
+							'You’ll also unlock advanced features that make it easy to build and grow your site.'
+						) }
+					</p>
+				</>
+			) : (
+				<p>{ translate( 'See and compare the features available on each WordPress.com plan.' ) }</p>
+			) }
+		</div>
+	);
+};
+
+const JetpackAppPlans: React.FC< JetpackAppPlansProps > = ( { paidDomainName, redirectTo } ) => {
+	const planSlug = useSelector( ( state: AppState ) =>
+		getPlanSlug( state, getPlan( PLAN_FREE )?.getProductId() || 0 )
+	) as string | null;
+	const plansLoaded = Boolean( planSlug );
+
+	const handleRedirect = ( planId?: number ) => {
+		if ( redirectTo ) {
+			const cartItemParam = planId ? `?plan_id=${ planId }` : '';
+			window.location.href = `${ redirectTo }${ cartItemParam }`;
+		}
+	};
+
+	const onUpgradeClick = ( cartItems?: MinimalRequestCartProduct[] | null | undefined ) => {
+		const productSlug = getPlanCartItem( cartItems )?.product_slug;
+
+		if ( ! productSlug ) {
+			return handleRedirect();
+		}
+
+		const plan = getPlan( productSlug ) as Plan;
+
+		handleRedirect( plan.getProductId() );
+	};
+
+	return (
+		<Main className="jetpack-app-plans">
+			<QueryPlans />
+			{ plansLoaded ? (
+				<>
+					<Header paidDomainName={ paidDomainName } />
+					<PlansFeaturesMain
+						paidDomainName={ paidDomainName }
+						intent="plans-jetpack-app-site-creation"
+						isInSignup={ true }
+						intervalType="yearly"
+						onUpgradeClick={ onUpgradeClick }
+						plansWithScroll={ false }
+						removePaidDomain={ handleRedirect }
+						hidePlanTypeSelector={ true }
+						hidePlansFeatureComparison={ true }
+					/>
+				</>
+			) : (
+				<div className="plans__loading">
```

Comment:
Might want to change some classnames to fit our [coding guidelines](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md#classes). The biggest deal here is that this classname might be in use already and we might need a proper prefix that fits the context, like `jetpack-app__plans-` or `jetpack-app-plans__`.